apps/web/app/queries/stacks/use-stacks-pending-transactions.ts

```diff
@@ -0,0 +1,66 @@
+import { useCallback, useMemo } from 'react';
+
+import { MempoolTransaction } from '@stacks/stacks-blockchain-api-types';
+import { useQueries } from '@tanstack/react-query';
+import { useStacksClient } from '~/queries/stacks/stacks-client';
+
+import { createGetAddressMempoolTransactionsQueryOptions } from '@leather.io/query';
+import { isUndefined } from '@leather.io/utils';
+
+import { useGetTransactionByIdListQuery } from '../transaction/transactions-by-id.query';
+
+export function useGetAddressMempoolTransactionsQueries(addresses: string[]) {
+  const client = useStacksClient();
+  const queries = useQueries({
+    queries: addresses.map(address => ({
+      ...createGetAddressMempoolTransactionsQueryOptions({ address, client }),
+    })),
+  });
```

Comment:
Why not use `combine` here?

Reply:
This code reimplements a feature included in the library https://tanstack.com/query/latest/docs/framework/react/reference/useQueries#combine